config/detekt/detekt.yml

```diff
@@ -5,14 +5,21 @@ config:
 
 complexity:
   LongParameterList:
-    ignoreAnnotated: ['javax.inject.Inject', 'javax.annotation.Generated']
+    ignoreAnnotated: ['javax.inject.Inject', 'javax.annotation.Generated', 'Composable']
 
 formatting:
   android: true
   autoCorrect: false
 
+# See: https://detekt.dev/docs/introduction/compose/
+naming:
+  FunctionNaming:
+    ignoreAnnotated: ['Composable']
+
 style:
   ForbiddenComment:
     values: ['FIXME:', 'STOPSHIP:']
   MagicNumber:
     ignoreEnums: true
+  UnusedPrivateMember:
+    allowedNames: '(_|ignored|expected|serialVersionUID)|([A-Za-z0-9]*Preview)'
```

Comment:
@ParaskP7 I'd love to get your feedback on these changes in `detekt.yml`.

The driving guide behind them is the official Detekt [Configuration for Compose](https://detekt.dev/docs/introduction/compose/) guide.

Not everything from there works as described but with the help of the information I got from that article I could wrangle a few customizations that allow the standard Detekt config work together with tweaks for Compose.

Thanks in advance 🙇

Reply:
👋 @ovitrif and thank you so much for the ping! 🙇 

I find that the changes you did expected, so it is a 👍 from my side as I was also expecting you to start having this kind of problems with Detekt and Compose as soon as you start writing more and more Compose code. 🥇 

The one thing I would suggest you to do before proceeding into merging those is:
- Check the Compose related configuration for this [detekt.yml](https://github.com/woocommerce/woocommerce-android/blob/trunk/config/detekt/detekt.yml) file on the WCAndroid side, they already did this Detekt for Compose update on their side to overcome these issues (see [relevant PR](https://github.com/woocommerce/woocommerce-android/pull/6524)). Look for `Composable` and `ignoreAnnotated` when searching for keywords.
- As such, and based on my search on the above, I am not sure if you can replace the `UnusedPrivateMember. allowedNames.regex` configuration with `UnusedPrivateMember. ignoreAnnotated. Preview`.
- Also, I am not sure if you want to add the below two extra configurations: 
    - `LongMethod. ignoreAnnotated. Composable`
    - `MagicNumber. ignoreAnnotated. Composable`
- An additional comment from my side would be to move the `# See: https://detekt.dev/docs/introduction/compose/` documentation comment on top of this `detekt.yml` file and below the existing `# Default Config: ...` documentation comment since this Compose documentation comment will be applied to all sections, not only `naming`. Wdyt? 🤔

Reply:
Thank you @ParaskP7 🥇 🙇 ,

I'll investigate all points you listed and come back with a resolution 👍

Reply:
👋 @ovitrif !

I just managed to create this [Detekt related PR](https://github.com/wordpress-mobile/WordPress-Android/pull/16949) (a very straightforward one), which updates Detekt to `1.19.0`. This [update](https://github.com/detekt/detekt/releases/tag/v1.19.0), amongst others includes the following:

```
Specifically, we've shipped some features that will allow you to better adapt detekt to run on codebases
that are using JetPack compose with features such as ignoreAnnotated and ignoreFunction.
```

Which means, if I am correct, that you could now definitely use the `UnusedPrivateMember. ignoreAnnotated. Preview` syntax instead of the `UnusedPrivateMember. allowedNames.regex` one. I hope this would unblock you. 

Thus, I shamelessly added you as a reviewer so that we could manage to merge this `1.19.0` update first (and as quick as possible), which you could then potentially use to update the `detekt.yml` on this PR of yours with the `Composable` related configuration, especially on the `ignoreAnnotated` and `ignoreFunction` side.

I hope that helps! 🙏

Reply:
👋 @ovitrif and FYI that this this [Detekt related PR](https://github.com/wordpress-mobile/WordPress-Android/pull/16949) that I mentioned yesterday, which updates Detekt to `1.19.0` is now merge to `trunk`. I recommend you update this PR of yours with `trunk` and then apply the suggestions we discussed afterwards. 💯

Reply:
Awesome work & timing @ParaskP7 🏅 , thanks to the great work you did with the Detekt upgrade to `1.19.0` I could apply all the suggestions and mix our `detekt.yml` with the WC one with regards to the Compose configuration 🎉 🚀

Reply:
Thank you so much for applying the suggestions @ovitrif ! ❤️ 

I did take a look at 0ebadc4adbdfb2451000b45e368bb9659155c04b and everything looks good, it is a 👍 on the Detekt side! 🚀 🟢

Reply:
👋 @ovitrif and another quick FYI for you that this `1.21.0` [Detekt related PR](https://github.com/wordpress-mobile/WordPress-Android/pull/16961) was just merged to `trunk` as well. Maybe you might want to build your Detekt Compose changes on top of that, but in terms of Compose rules for Detekt, nothing really changed.

And... with that we are now using the latest version of Detekt for WPAndroid! 🎉

Reply:
Party time 🍺 🍻 🎊 🥳 🎉 . Thank you @ParaskP7 🙇 !!!